### PR TITLE
Move core assessments and visually separate tasks in score report

### PR DESCRIPTION
**Pull request by Emily-ejag**
## Proposed changes

On the Score Report move "Word, Sentence, and Phoneme" first, then gray out cell that belong to the other task because those are not validated.

On Progress Report only move the validated tasks first.

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [x] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [x] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [ ] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [x] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [ ] I have linked relevant issues (if any)

### CODE REVIEW

File: src/pages/ScoreReport.vue [hunk 1]
```diff
@@ -1079,6 +1095,15 @@ const scoreReportColumns = computed(() => {
...
       colField = `scores.${taskId}.rawScore`;
     }
+
+    let backgroundColor = '';
+
+    if (taskId === 'pa' || taskId === 'sre' || taskId === 'swr') {
+      backgroundColor = 'transparent';
+    } else {
+      backgroundColor = '#E6E6E6';
+    }
```
Comment (ksmontville) [inline src/pages/ScoreReport.vue:1105]:
```suggestion
    if (priorityTasks.includes(taskId) {
      backgroundColor = 'transparent';
    } else {
      backgroundColor = '#E6E6E6';
    }
```

File: src/pages/ScoreReport.vue [hunk 2]
```diff
@@ -1106,16 +1132,20 @@ const allTasks = computed(() => {
...
 
 const sortedTaskIds = computed(() => {
-  const res = Object.keys(computeAssignmentAndRunData.value.runsByTaskId).toSorted((p1, p2) => {
-    if (Object.keys(taskDisplayNames).includes(p1) && Object.keys(taskDisplayNames).includes(p2)) {
-      return taskDisplayNames[p1].order - taskDisplayNames[p2].order;
-    } else {
-      return -1;
-    }
+  const runsByTaskId = computeAssignmentAndRunData.value.runsByTaskId;
+  const specialTaskIds = ['swr', 'sre', 'pa'].filter((id) => Object.keys(runsByTaskId).includes(id));
```
Comment (ksmontville) [inline src/pages/ScoreReport.vue:1136]:
```suggestion
  const specialTaskIds = priorityTasks.filter((id) => Object.keys(runsByTaskId).includes(id));
```

File: src/pages/ScoreReport.vue [hunk 3]
```diff
@@ -1106,16 +1132,20 @@ const allTasks = computed(() => {
...
+  remainingTaskIds.sort((p1, p2) => {
+    return taskDisplayNames[p1].order - taskDisplayNames[p2].order;
   });
-  return res;
+
+  const sortedIds = specialTaskIds.concat(remainingTaskIds);
+  return sortedIds;
 });
 
+console.log('sorted tasks ids', sortedTaskIds);
```
Comment (ksmontville) [inline src/pages/ScoreReport.vue:1147]: Remove when no longer needed for debugging.

File: src/pages/ScoreReport.vue [hunk 4]
```diff
@@ -1065,7 +1065,23 @@ const scoreReportColumns = computed(() => {
...
+    if (sortedTasks.includes(task)) {
+      orderedTasks.push(task);
+    }
+  }
+
+  for (const task of sortedTasks) {
+    if (!priorityTasks.includes(task)) {
+      orderedTasks.push(task);
+    }
+  }
```
Comment (ksmontville) [inline src/pages/ScoreReport.vue:1082]:
```suggestion
const filteredPriorityTasks = priorityTasks.filter( task => sortedTask.includes(task))
const filteredSortedTasks = sortedTasks.filter( task => !priorityTasks.includes(task))

const orderedTasks = filteredPriorityTasks.concat(filteredSortedTasks)
```

**Review by ksmontville — APPROVED** [review]

File: src/pages/ScoreReport.vue [hunk 5]
```diff
@@ -1106,16 +1125,20 @@ const allTasks = computed(() => {
...
+  remainingTaskIds.sort((p1, p2) => {
+    return taskDisplayNames[p1].order - taskDisplayNames[p2].order;
   });
-  return res;
+
+  const sortedIds = specialTaskIds.concat(remainingTaskIds);
+  return sortedIds;
 });
 
+console.log('sorted tasks ids', sortedTaskIds);
```
Comment (Emily-ejag) [inline src/pages/ScoreReport.vue:1140]:
```suggestion

```

**Pull request merged**